Mathlib/Topology/PartitionOfUnity.lean

```diff
@@ -575,4 +622,60 @@ theorem exists_isSubordinate [NormalSpace X] [ParacompactSpace X] (hs : IsClosed
   let ⟨f, hf⟩ := BumpCovering.exists_isSubordinate hs U ho hU
   ⟨f.toPartitionOfUnity, hf.toPartitionOfUnity⟩
 
+/-- If `X` is a locally compact T2 topological space and `U` is a locally finite open covering of a
+compact set `s`, then there exists a `PartitionOfUnity ι X s` that is subordinate to `U`. -/
+theorem exists_isSubordinate_of_locallyFinite_t2space [LocallyCompactSpace X] [T2Space X]
+    (hs : IsCompact s) (U : ι → Set X) (ho : ∀ i, IsOpen (U i)) (hf : LocallyFinite U)
+    (hU : s ⊆ ⋃ i, U i) : ∃ f : PartitionOfUnity ι X s, f.IsSubordinate U ∧
+    ∀ i, HasCompactSupport (f i) :=
+  let ⟨f, hfsub, hfcp⟩ :=
+    BumpCovering.exists_isSubordinate_hasCompactSupport_of_locallyFinite_t2space hs U ho hf hU
+  ⟨f.toPartitionOfUnity, hfsub.toPartitionOfUnity, fun i => IsCompact.of_isClosed_subset (hfcp i)
+    isClosed_closure <| closure_mono (f.support_toPartitionOfUnity_subset i)⟩
+
 end PartitionOfUnity
+
+/-- A variation of Urysohn's lemma. In a locally compact T2 space `X`, for a compact set `t` and a
+finite family of open sets `{s i}_i` such that `t ⊆ ⋃ i, s i`, there is a family of compactly
+supported continuous functions `{f i}_i` supported in `s i`, `∑ i, f i x = 1` on `t` and
+`0 ≤ f i x ≤ 1`. -/
+theorem exists_continuous_sum_one_of_isOpen_isCompact [T2Space X] [LocallyCompactSpace X]
+    {n : ℕ} {t : Set X} {s : Fin n → Set X} (hs : ∀ (i : Fin n), IsOpen (s i)) (htcp : IsCompact t)
+    (hst : t ⊆ ⋃ i, s i) : ∃ f : Fin n → C(X, ℝ), (∀ (i : Fin n), tsupport (f i) ⊆ s i) ∧
+    EqOn (∑ i, f i) 1 t ∧ (∀ (i : Fin n), ∀ (x : X), f i x ∈ Icc (0 : ℝ) 1) ∧ (∀ (i : Fin n),
+    HasCompactSupport (f i)) := by
+  have hlf : LocallyFinite s := by
+    intro x
+    use univ
+    refine ⟨univ_mem, ?_⟩
+    exact toFinite {i | (s i ∩ univ).Nonempty}
+  obtain ⟨f, hfsub, hfcp⟩ := PartitionOfUnity.exists_isSubordinate_of_locallyFinite_t2space htcp s
+    hs hlf hst
+  use f.toFun
+  refine ⟨fun i ↦ hfsub i, ?_, ?_, fun i => hfcp i⟩
+  · intro t ht
+    simp only [Finset.sum_apply, Pi.one_apply]
+    have h := f.sum_eq_one' t ht
+    simp at h
+    rw [finsum_eq_sum (fun i => (f.toFun i) t)
+      (Finite.subset finite_univ (subset_univ (support fun i ↦ (f.toFun i) t)))] at h
+    simp only [Finite.toFinset_setOf, ne_eq] at h
+    rw [← h, ← Finset.sum_subset]
+    · exact Finset.filter_subset (fun x ↦ ¬(f.toFun x) t = 0) Finset.univ
+    · intro x _ hnx
+      simp only [Finset.mem_filter, Finset.mem_univ, true_and, Decidable.not_not] at hnx
+      exact hnx
+  intro i x
+  constructor
+  · exact f.nonneg' i x
+  by_cases h0 : f.toFun i x = 0
+  · rw [h0]
+    exact zero_le_one
+  rw [← Finset.sum_singleton (f.toFun ·  x) i]
+  apply le_trans _ (f.sum_le_one' x)
+  rw [finsum_eq_sum (f.toFun ·  x) (by exact toFinite (support fun x_1 ↦ (f.toFun x_1) x))]
```

Comment:
```suggestion
  rw [finsum_eq_sum (f · x) (by exact toFinite (support (f · x)))]
```

Reply:
I had to revert this because it gave an error